assets/js/services.js

```diff
@@ -48,16 +48,78 @@ storefrontApp.service('pricingService', ['$http', function ($http) {
 	}
 }]);
 
-storefrontApp.service('catalogService', ['$http', function ($http) {
+storefrontApp.service('catalogService', ['$http', '$localStorage', function($http, $localStorage) {
```

Comment:
Define separate service 'compareProductService' with follow methods 'getProductsCount', 'removeProduct(id)', 'addProduct(product)', 'Product[] getProducts'
Move 'getProductProperties'  method to ` product-compare.js` controller